server.go

```diff
@@ -595,3 +600,14 @@ func suitableMethods(typ reflect.Type, reportErr bool) map[string]*methodType {
 	}
 	return methods
 }
+
+// GetRequestSender gets current request sender from RPC service's function context
+// If it is nil, then consider this call as local.
+func GetRequestSender(ctx context.Context) *peer.ID {
+	v := ctx.Value(ContextKeyRequestSender)
+	if v == nil {
+		return nil
+	}
+	p := v.(peer.ID)
```

Comment:
`p, ok := ...` always good to assume the program can crash.

Also, https://pkg.go.dev/github.com/libp2p/go-libp2p-core@v0.8.5/peer#ID.Validate .